cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -170,9 +178,11 @@ func ensureLayoutCompatibility(gadgetLayout *gadget.LaidOutVolume, diskLayout *p
 		return fmt.Errorf("disk ID %q doesn't match gadget volume ID %q", diskLayout.ID, gadgetLayout.Volume.ID)
 	}
 
-	// Check if all existing device partitions are also in gadget
+	// Check if all existing device partitions, except those that we added
+	// ourselves, are also in gadget
+	toRemove := partitionListCreatedPartitions(diskLayout)
```

Comment:
I don't understand this change, ubuntu-data is in gadget

Reply:
This was in place for the previous approach where we deleted all partitions with the attribute set, regardless of being known or not (thus the sanity check ignored them all because they would be deleted anyway). In the new approach we don't need to exclude partitions with the attribute so this needs to be fixed. We'll also need an exception for the growable data partition size later in PR #8253